File: src/algorithms/detect-capital-use.js
/**
 * Given a string word, returns true if the usage of capitals is correct.
 * 
 * We define the usage of capitals in a word to be right when one of the
 * following cases holds:
 *     • All letters in the word are capitals ("USA").
 *     • All letters in the word are not capitals ("leetcode").
 *     • Only the first letter in this word is capital ("Google").
 * 
 * @param {string} word - A string of English letters
 * @returns {boolean} - Returns true if capitalization rules are followed
 */
function detectCapitalUse (word) {
  return word === word.toUpperCase() ||
         word === word.toLowerCase() ||
         word === word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
}

File: src/algorithms/detect-capital-use.py
def detect_capital_use(word):
    """We define the usage of capitals in a word to be right when one of the 
    following cases holds:
    • All letters in this word are capitals, like "USA".
    • All letters in this word are not capitals, like "leetcode".
    • Only the first letter in this word is capital, like "Google".
    Given a string word, return true if the usage of capitals in it is right.

    Args:
        word (str): A string of English letters.

    Returns:
        Boolean: Returns True if the usage of capitals in word is right.
    """
    return (word == word.upper() or
            word == word.lower() or
            word == word[0].upper() + word[1:].lower())
